src/t_string.c

```diff
@@ -486,6 +487,9 @@ void setrangeCommand(client *c) {
 
         /* Create a copy when the object is shared or encoded. */
         kv = dbUnshareStringValueByLink(c->db, c->argv[1], kv, link);
+
+        newLen = max(oldLen, (int64_t) (offset + value_len));
+        updateKeysizesHist(c->db, getKeySlot(c->argv[1]->ptr), OBJ_STRING, oldLen, newLen);
```

Comment:
maybe we should explain here why we call updateKeysizesHist() in advance, or add a comment to dbAddByLink() to indicate that the length has been updated there.